feature-libs/cart/main/root/components/mini-cart/mini-cart-component.service.ts

```diff
@@ -0,0 +1,120 @@
+import { Injectable } from '@angular/core';
+import {
+  AuthService,
+  BASE_SITE_CONTEXT_ID,
+  FacadeFactoryService,
+  SiteContextParamsService,
+  StatePersistenceService,
+  StorageSyncType,
+  UnifiedInjector,
+} from '@spartacus/core';
+import { combineLatest, Observable, of } from 'rxjs';
+import {
+  distinctUntilChanged,
+  filter,
+  map,
+  startWith,
+  switchMap,
+  takeWhile,
+} from 'rxjs/operators';
+import { ActiveCartFacade } from '../../facade/active-cart.facade';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class MiniCartComponentService {
+  constructor(
+    protected activeCartFacade: ActiveCartFacade,
+    protected authService: AuthService,
+    protected statePersistenceService: StatePersistenceService,
+    protected siteContextParamsService: SiteContextParamsService,
+    protected injector: UnifiedInjector,
+    protected facadeFactoryService: FacadeFactoryService
+  ) {}
+
+  /**
+   * This function supports lazy loading of the cart functoinality's code. We only call
+   * the activeCartFacade if we know there is actually a cart.
+   * Without a cart, we can return a default value and
+   * avoid loading the cart library code.
+   */
+  getQuantity(): Observable<number> {
+    return this.watchUntilCartFacadeRequired().pipe(
+      switchMap((userHasCart) => {
+        if (userHasCart) {
+          return this.activeCartFacade.getActive().pipe(
+            startWith({ deliveryItemsQuantity: 0 }),
+            map((cart) => cart.deliveryItemsQuantity || 0)
+          );
+        } else {
+          return of(0);
+        }
+      })
+    );
+  }
+
+  /**
+   * This function supports lazy loading of the cart functoinality's code. We only call
+   * the activeCartFacade if we know there is actually a cart.
+   * Without a cart, we can return a default value and
+   * avoid loading the cart library code.
+   */
+  getTotalPrice(): Observable<string> {
```

Comment:
except these 2 get functions, others can be `protected`, right?

Reply:
I guess they can.  Their purpose is really to help the 2 get functions accomplish their task.

Reply:
I made the protected